packages/deployer/src/utils/compiler.ts

```diff
@@ -7,6 +7,51 @@ import { constants } from './constants';
 import { fsWrapper } from './fs_wrapper';
 import { ContractArtifact, ContractSources } from './types';
 
+/**
+* Constructs a system-wide unique identifier for a source file.
+* @param directoryNamespace Namespace of the source file's root contract directory.
+* @param sourceFilePath Path to a source file, relative to contractBaseDir.
+* @return sourceFileId A system-wide unique identifier for the source file.
+*/
+export function constructSourceFileId(directoryNamespace: string, sourceFilePath: string): string {
+    let namespacePrefix:string = "";
+    if(directoryNamespace != "") {
+        namespacePrefix = "/" + directoryNamespace;
```

Comment:
Template strings. Please check all new code for this. Strict equality, etc... too. Will stop commenting on them.